substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrEvents.java

```diff
@@ -56,11 +83,20 @@ private static long getEventTypeId(String name) {
                     return type.getId();
                 }
             }
-            return 0;
+
+            String exceptionMessage = "Event " + name + " is not found!";
+            String mostSimilarEvent = getMostSimilarEvent(name);
+            if (mostSimilarEvent != null) {
+                exceptionMessage += " Most similar event is " + mostSimilarEvent + ".";
+            }
+            exceptionMessage += " Take a look at 'metadata.xml' to see all available events.";
+
+            VMError.shouldNotReachHere(exceptionMessage);
         } catch (Exception ex) {
             VMError.shouldNotReachHere(ex);
```

Comment:
throw here to interrupt the control flow for the IDEs.

Reply:
Done.